pkg/metrics/metrics.go

```diff
@@ -70,7 +70,7 @@ func (mc MachineCollector) Describe(ch chan<- *prometheus.Desc) {
 func (mc MachineCollector) collectMachineMetrics(ch chan<- prometheus.Metric) {
 	machineList, err := mc.listMachines()
 	if err != nil {
-		ScrapeFailedCounter.With(prometheus.Labels{"kind": "Machine-count", "reason": err.Error()}).Inc()
+		ScrapeFailedCounter.With(prometheus.Labels{"kind": "Machine-count"}).Inc()
```

Comment:
On second thought, this metric should be a [gauge](https://github.com/openshift/machine-api-operator/pull/335#discussion_r304289521) rather than a counter. See https://prometheus.io/docs/instrumenting/writing_exporters/#metrics-about-the-scrape-itself.

Reply:
@simonpasquier 
> These should be exposed as gauges as they’re about an event

Looks like guage is suggested for the metric which are for specific scrape and since this metrics is not ablut any specific scrape rather reporting the overall status till any moment, do you think Guwague makes sense? i am not sure.

Reply:
It is still relevant IMO. This code runs in some background loop, it is only executed when Prometheus (or something else) scrapes the metrics endpoint.

Reply:
agree that it runs when prometheus (or something else) scrapes the metrics endpoint but the metric value is not "about this particular scrape(event)", rather it is cumulative info about all such events that happened till any moment.

Reply:
Think about alerting on this metric: you're interesting to know if several scrapes fail in a row (as in general your other alerts should be resilient to occasional scrape failures).
With a gauge, the alert is `mapi_collector_up == 0` combined with a `for` condition equal to a few times the scrape interval.
With a counter you would have to expose `mapi_scrape_failures_total` and `mapi_scrapes_total` then query `rate(mapi_scrape_failures_total[5m]) / rate(mapi_scrapes_total[5m] > x` with `x` being harder to determine. And this alert may not resolved immediately when the scrapes are successful again.

Reply:
@simonpasquier thanks! make sense!